### ISSUE TITLE
Killing worker process shuts down the entire dramatiq queue with exit code 0

### ISSUE THREAD

**Issue by StarLightPL** [Bogdanp/dramatiq#376]
## Checklist

* [X] Does your title concisely summarize the problem?
* [X] Did you include a minimal, reproducible example?
* [X] What OS are you using?
* [X] What version of Dramatiq are you using?
* [X] What did you do?
* [X] What did you expect would happen?
* [X] What happened?

## What OS are you using?
```
$ cat /etc/redhat-release
CentOS Linux release 7.8.2003 (Core)
```

## What version of Dramatiq are you using?

```
$ python -c 'import dramatiq; print(dramatiq.__version__)'
1.9.0
$ redis-server --version
Redis server v=3.2.12 sha=00000000:0 malloc=jemalloc-3.6.0 bits=64 build=7897e7d0e13773f
```

## Issue description

OOMKiller killed one of dramatiq worker processes, which resulted in whole dramatiq queue shutting down, and more importantly, with clean exit code (0) which prevented supervisor from restarting it. Exit code 0 means a clean (expected) exit. This is a severe issue, because an exit like this will lead to the whole task processing going down and not being revived by external tools, like supervisor.

## What did you expect would happen?

1. In case of abnormal termination like this one, dramatiq should exit with non-zero exit code, indicating an error.
2. Death of one worker shouldn't bring the whole queue down. Worker should be restarted by the main process, not interrupt the whole queue.

## How to reproduce

Here's a small `sleepy_actor.py` to reproduce the issue:
```python
import os
import time

import dramatiq
from dramatiq.brokers.redis import RedisBroker

dramatiq.set_broker(RedisBroker(
    namespace='mynamespace:dramatiq',
    url='redis://localhost:6379/0'
))


@dramatiq.actor()
def victim() -> None:
    print(f"Waiting to be killed, process {os.getpid()}")
    time.sleep(900)

```

Let's make sure redis is empty:
```
$ redis-cli
127.0.0.1:6379> keys *
(empty list or set)
```

Run the dramatiq command line:
```
$ dramatiq -p 2 -t 2 --use-spawn sleepy_actor
[2021-01-22 15:09:25,459] [PID 10531] [MainThread] [dramatiq.MainProcess] [INFO] Dramatiq '1.9.0' is booting up.
[2021-01-22 15:09:25,458] [PID 10577] [MainThread] [dramatiq.WorkerProcess(0)] [INFO] Worker process is ready for action.
[2021-01-22 15:09:25,458] [PID 10578] [MainThread] [dramatiq.WorkerProcess(1)] [INFO] Worker process is ready for action.
[2021-01-22 15:09:25,546] [PID 10593] [MainThread] [dramatiq.ForkProcess(0)] [INFO] Fork process 'dramatiq.middleware.prometheus:_run_exposition_server' is ready for action.
```

Now send the task into queue:
```
$ python
Python 3.7.2 (default, Dec 16 2020, 10:56:41)
[GCC 4.8.5 20150623 (Red Hat 4.8.5-39)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import sleepy_actor
>>> sleepy_actor.victim.send()
Message(queue_name='default', actor_name='victim', args=(), kwargs={}, options={'redis_message_id': 'f5b899b5-df23-4c0d-b397-7e0d5fd9470a'}, message_id='2c4d87a3-a5f0-4e29-94c0-5fb7f3b88701', message_timestamp=1611324765754)
>>>
```

Now we kill the worker process from the commandline
```
$ kill -9 10578
```

and let's see what happened in the command line utility output:

```
Waiting to be killed, process 10578
[2021-01-22 15:13:48,850] [PID 10531] [MainThread] [dramatiq.MainProcess] [CRITICAL] Worker with PID 10578 exited unexpectedly (code -9). Shutting down...
[2021-01-22 15:13:48,850] [PID 10577] [MainThread] [dramatiq.WorkerProcess(0)] [INFO] Stopping worker process...
[2021-01-22 15:13:48,850] [PID 10593] [MainThread] [dramatiq.ForkProcess(0)] [INFO] Stopping fork process...
[2021-01-22 15:13:49,790] [PID 10577] [MainThread] [dramatiq.worker.Worker] [INFO] Shutting down...
[2021-01-22 15:13:50,961] [PID 10577] [MainThread] [dramatiq.worker.Worker] [INFO] Worker has been shut down.
$ echo $?
0
```

It exited gracefully while it should either exit with error, or restart the worker process and not exit at all.

**Comment by Bogdanp** [comment]
This was a regression in 1.9 and should be fixed in 1.10 (#333, #334) so please upgrade. It's intentional that dramatiq shuts down when workers die and the idea is to let the service manager (systemd, supervisor, etc) handle restarting .

**Comment by StarLightPL** [comment]
We'll upgrade, thanks!
